vault/provider.go

```diff
@@ -869,17 +881,22 @@ func ReadContextWrapper(f schema.ReadContextFunc) schema.ReadContextFunc {
 // wrapped schema.CreateContextFunc.
 func MinVersionCheckWrapper(f schema.CreateContextFunc, minVersion string) schema.CreateContextFunc {
 	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-		client, e := provider.GetClient(d, meta)
-		if e != nil {
-			return diag.FromErr(e)
+		versionMap := map[string]*version.Version{
```

Comment:
I think if you change signature of the outer function to take a `version.Version` then you can probably drop the map altogether.

Reply:
I tried doing that, but unfortunately at the time of passing the version variables to the `CreateContext` function of a resource, the `VaultVersionXXX` version variables do not initialize, and are always being passed to this function as `nil`

Reply:
Updated after our discussion. Thanks!